File: CommandCPP/RemoteControl.cpp
/* Copyright (c) 2022 RodKingroo */
#include "RemoteControl.h"

RemoteControl::RemoteControl() 
{
	std::shared_ptr<Command> noCommand = std::make_shared<NoCommand>();
	for (int i = 0; i < 37; i++) {
		onCommands.push_back(noCommand);
		offCommands.push_back(noCommand);
	}
	
	undoCommand = noCommand;

}

void RemoteControl::setCommand(int slot, std::shared_ptr<Command> onCommand, std::shared_ptr<Command> offCommand)
{
	onCommands[slot] = onCommand;
	offCommands[slot] = offCommand;
}

void RemoteControl::onButtonPress(int slot) {
	onCommands[slot]->execute();
	undoCommand = onCommands[slot];
}

void RemoteControl::offButtonPress(int slot) {
	offCommands[slot]->execute();
	undoCommand = offCommands[slot];

}

void RemoteControl::undoButtonPress() {
	undoCommand->undo();
}

File: CommandPython/RemoteControl.py
# Copyright (c) 2022 RodKingroo

from Command import *
from NoCommand import *

class RemoteControl:
    def __init__(self):
        self.undoCommand = NoCommand()
        self.onCommands = []
        self.offCommands = []
        for i in range(47):
            self.onCommands.append(self.undoCommand)
            self.offCommands.append(self.undoCommand)
            
    def setCommand(self, slot, onCommand, offCommand):
        self.onCommands[slot] = onCommand
        self.offCommands[slot] = offCommand
        
    def onButtonPress(self, slot):
        self.onCommands[slot].execute()
        self.undoCommand = self.onCommands[slot]
        
    def offButtonPress(self, slot):
        self.offCommands[slot].execute()
        self.undoCommand = self.offCommands[slot]
        
    def undoButtonPress(self):
        self.undoCommand.undo()